crates/core/tedge_api/src/entity_store.rs

```diff
@@ -529,38 +522,17 @@ impl EntityStore {
             twin_data: Map::new(),
         };
 
-        // device is affected if it was previously registered and was updated
-        // (i.e. EntityMetadata has changed)
-        let previous = self.entities.entry(topic_id.clone());
-        match previous {
-            Entry::Occupied(mut occupied) => {
-                // if there is no change, no entities were affected
-                let existing_entity = occupied.get().clone();
-
-                let mut merged_other = existing_entity.other.clone();
-                merged_other.extend(entity_metadata.other.clone());
-                let merged_entity = EntityMetadata {
-                    twin_data: existing_entity.twin_data.clone(),
-                    other: merged_other,
-                    ..entity_metadata
-                };
-
-                if existing_entity == merged_entity {
-                    return Ok(vec![]);
```

Comment:
Extracting this code under an `EntityTree::insert` method has been unexpectedly complicated by this return which ignores the vector of entities affected so far.

The contract of this method is not fully clear to me.